src/MessageTranslator.cpp

```diff
@@ -0,0 +1,281 @@
+    uriPtr->mutable_entity()->CopyFrom(uEntity);
+    uriPtr->mutable_resource()->CopyFrom(uResource);
+
+    //build Attribute using the URI
+    UUID const &generatedUUID = uprotocol::uuid::Uuidv8Factory::create();
+    UMessageType const type = convertSomeipToUMessageType(someIpMessage->get_message_type());
+    UAttributesBuilder builder(*uriPtr ,generatedUUID, type, uprotocol::v1::UPriority::UPRIORITY_CS0);
+    std::shared_ptr<UAttributes> attributes = std::make_shared<UAttributes>(builder.build());
+
+    //build Payload
+    std::shared_ptr<UPayload> uPayload = buildUPayload(someIpMessage);
+
+    //build UMessage using attribute and payload
+    std::shared_ptr<UMessage> umsg = std::make_shared<UMessage>();
+    umsg->setPayload(*uPayload);
+    umsg->setAttributes(*attributes);
+
+    return umsg;
+
+}
+
+/**
+  * @brief @return @see @ref MessageTranslator::translateSomeipToUMsgForSubscriptionAck()
+  * @param[in] uEntity @see @ref MessageTranslator::translateSomeipToUMsgForSubscriptionAck()
+  * @param[in] uAuthority @see @ref MessageTranslator::translateSomeipToUMsgForSubscriptionAck()
+  * @param[in] uResource @see @ref MessageTranslator::translateSomeipToUMsgForSubscriptionAck()
+  */
+std::shared_ptr<UMessage> MessageTranslator::translateSomeipToUMsgForSubscriptionAck(UEntity const &uEntity, UAuthority const &uAuthority ,UResource const &uResource) const {
+    // build URI
+    std::shared_ptr<UUri> uriPtr = std::make_shared<UUri>();
+    uriPtr->mutable_authority()->CopyFrom(uAuthority);
+    uriPtr->mutable_entity()->CopyFrom(uEntity);
+    uriPtr->mutable_resource()->CopyFrom(uResource);
+
+    // build Attribute
+    UUID const &generatedUUID = uprotocol::uuid::Uuidv8Factory::create();
+    UMessageType const type = UMessageType::UMESSAGE_TYPE_UNSPECIFIED;
+    UAttributesBuilder builder(*uriPtr, generatedUUID, type, uprotocol::v1::UPriority::UPRIORITY_CS0);
+    std::shared_ptr<UAttributes> attributes = std::make_shared<UAttributes>(builder.build());
+
+    // build UMessage using attribute and payload
+    std::shared_ptr<UMessage> umsg = std::make_shared<UMessage>();
+    umsg->setAttributes(*attributes);
+
+    return umsg;
+}
+
+/**
+  * @brief @return @see @ref MessageTranslator::translateSomeipToUMsgForResponse()
+  * @param[in] someIpMessage @see @ref MessageTranslator::translateSomeipToUMsgForResponse().
+  * @param[in] uri @see @ref MessageTranslator::translateSomeipToUMsgForResponse().
+  * @param[in] uuid @see @ref MessageTranslator::translateSomeipToUMsgForResponse().
+  */
+std::shared_ptr<UMessage> MessageTranslator::translateSomeipToUMsgForResponse(
+    std::shared_ptr<message> const &someIpMessage,
+    std::shared_ptr<uprotocol::utransport::UMessage> originalRequestMsg) const {
+
+    std::shared_ptr<UMessage> uMsg = std::make_shared<UMessage>();
+
+    //UUID const &generatedUUID = uprotocol::uuid::Uuidv8Factory::create(); //TODO : Remove this comment once zenoh uTransport fix the issue.
```

Comment:
UUID in a response needs to always be a generic new UUID and the request id is populated in reqid attribute